apps/sms/views/conversation/js/attachment_renderer.js

```diff
@@ -204,8 +201,18 @@
               // It's essential to remember real image URL to revoke it later
               attachmentContainer.dataset.thumbnail = data.thumbnail.url;
             }
-          });
-      }.bind(this));
+
+            var fileNode = contentContainer.querySelector('.file-name');
+            if (this._attachment.name) {
+              fileNode.removeAttribute('data-l10n-id');
+              fileNode.textContent = this._attachment.name.slice(
+                this._attachment.name.lastIndexOf('/') + 1
+              );
+            } else {
+              fileNode.setAttribute('data-l10n-id', 'unnamed-attachment');
+            }
+        });
```

Comment:
following up on the previous discussion:

> @julienw: tell me what you think, but I also wonder if we couldn't use formatValue in getBaseMarkup to get the localized value ? getBaseMarkup would then return a Promise. So we wouldn't need to do some DOM operation after rendering. what do you think ?
> @zbraniecki: I don't know if I understand. Do you want getBaseMarkup to figure out if there is this._attachment.name and either get it's value from formatValue or do the sliceing on the name?

Yeah that was my idea; now I understand using data-l10n-id for the "unnamed" case is useful. So let's keep it like this, maybe we'll be able to simplify in a future patch.